ssl/statem/extensions.c

```diff
@@ -357,11 +357,12 @@ static int extension_is_relevant(SSL *s, unsigned int extctx,
  * found, or an internal error occurred. We only check duplicates for
  * extensions that we know about. We ignore others.
  */
-int tls_collect_extensions(SSL *s, PACKET *packet, unsigned int context,
+size_t tls_collect_extensions(SSL *s, PACKET *packet, unsigned int context,
```

Comment:
I prefer not to overload the return type to mean different things, i.e. I'd prefer a simple boolean (int) return type to indicate success or failure, with a separate `size_t *` to set the length. Incidentally the length should *definitely* be a size_t and not an int in order to match the types. I have seen far too many subtle bugs introduced by lengths swapping backwards and forwards between int and size_t. They should be consistent throughout. I don't want to have to redo all the size_t work I did in libssl just recently.